econml/inference/_inference.py

```diff
@@ -240,8 +243,11 @@ def effect_inference(self, X, *, T0, T1):
         e_stderr = self._prediction_stderr(cross_product(X, T1 - T0))
         d_y = self._d_y[0] if self._d_y else 1
         # d_t=1 here since we measure the effect across all Ts
+        # A dummy treatment names is passed in
         return NormalInferenceResults(d_t=1, d_y=d_y, pred=e_pred,
-                                      pred_stderr=e_stderr, inf_type='effect', **self._input_names)
+                                      pred_stderr=e_stderr, inf_type='effect',
+                                      feature_names=self._input_names["feature_names"],
```

Comment:
I don't like that this is accessing the `_input_names` dict directly. This hsould instead be accessing the public `cate_feature_names` and `cate_output_names`.

Reply:
Fixed!